x-pack/plugins/observability/public/components/shared/slo/slo_selector/slo_selector.tsx

```diff
@@ -0,0 +1,67 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { EuiComboBox, EuiComboBoxOptionOption } from '@elastic/eui';
+import { i18n } from '@kbn/i18n';
+import { debounce } from 'lodash';
+import React, { useEffect, useMemo, useState } from 'react';
+
+import { SLO } from '../../../../typings';
+import { useFetchSloList } from '../../../../hooks/slo/use_fetch_slo_list';
+
+interface Props {
+  onSelected: (slo: SLO) => void;
+}
+
+function SloSelector({ onSelected }: Props) {
+  const [options, setOptions] = useState<Array<EuiComboBoxOptionOption<string>>>([]);
+  const [selectedOptions, setSelected] = useState<Array<EuiComboBoxOptionOption<string>>>();
+  const [searchValue, setSearchValue] = useState<string>('');
+  const [loading, sloList] = useFetchSloList(searchValue);
+
+  useEffect(() => {
```

Comment:
This might not have to be in a `useEffect` / `setOptions` pattern.

I believe this can just be derived:

```suggestion
const options = !loading && sloList !== undefined ? sloList.results.map((slo) => ({ value: slo.id, label: slo.name })) : []
```